test/functional/connection.test.js

```diff
@@ -27,12 +41,12 @@ describe('Connection - functional', function () {
         isMonitoring = event instanceof ServerHeartbeatStartedEvent;
       });
 
-      return client
+      client
         .connect()
         .then(() => {
           expect(isMonitoring);
         })
-        .finally(() => client.close());
+        .finally(done);
```

Comment:
since you are rewriting this section to use a `done()` callback instead of the async return syntax, you don't need a finally anymore, you can delete this line and move the `done()` to the `.then` clause, to run after expect